framework/src/Volo.Abp.MongoDB/Volo/Abp/Domain/Repositories/MongoDbCoreRepositoryExtensions.cs

```diff
@@ -11,60 +11,56 @@ namespace Volo.Abp.Domain.Repositories
+        public static Task<IMongoDatabase> GetDatabaseAsync<TEntity>(this IBasicRepository<TEntity> repository, CancellationToken cancellationToken = default)
+            where TEntity : class, IEntity
         {
             return repository.ToMongoDbRepository().GetDatabaseAsync(cancellationToken);
         }
 
         [Obsolete("Use GetCollectionAsync method.")]
-        public static IMongoCollection<TEntity> GetCollection<TEntity, TKey>(this IBasicRepository<TEntity, TKey> repository)
-            where TEntity : class, IEntity<TKey>
+        public static IMongoCollection<TEntity> GetCollection<TEntity>(this IBasicRepository<TEntity> repository)
+            where TEntity : class, IEntity
         {
             return repository.ToMongoDbRepository().Collection;
         }
 
-        public static Task<IMongoCollection<TEntity>> GetCollectionAsync<TEntity, TKey>(this IBasicRepository<TEntity, TKey> repository, CancellationToken cancellationToken = default)
-            where TEntity : class, IEntity<TKey>
+        public static Task<IMongoCollection<TEntity>> GetCollectionAsync<TEntity>(this IBasicRepository<TEntity> repository, CancellationToken cancellationToken = default)
+            where TEntity : class, IEntity
         {
             return repository.ToMongoDbRepository().GetCollectionAsync(cancellationToken);
         }
 
         [Obsolete("Use GetMongoQueryableAsync method.")]
-        public static IMongoQueryable<TEntity> GetMongoQueryable<TEntity, TKey>(this IBasicRepository<TEntity, TKey> repository)
-            where TEntity : class, IEntity<TKey>
+        public static IMongoQueryable<TEntity> GetMongoQueryable<TEntity>(this IBasicRepository<TEntity> repository)
+            where TEntity : class, IEntity
         {
             return repository.ToMongoDbRepository().GetMongoQueryable();
         }
 
-        public static Task<IMongoQueryable<TEntity>> GetMongoQueryableAsync<TEntity, TKey>(this IBasicRepository<TEntity, TKey> repository, CancellationToken cancellationToken = default)
-            where TEntity : class, IEntity<TKey>
+        public static Task<IMongoQueryable<TEntity>> GetMongoQueryableAsync<TEntity>(this IBasicRepository<TEntity> repository, CancellationToken cancellationToken = default)
+            where TEntity : class, IEntity
         {
             return repository.ToMongoDbRepository().GetMongoQueryableAsync(cancellationToken);
         }
 
-        public static Task<IAggregateFluent<TEntity>> GetAggregateAsync<TEntity, TKey>(this IBasicRepository<TEntity, TKey> repository, CancellationToken cancellationToken = default)
-            where TEntity : class, IEntity<TKey>
+        public static async Task<IAggregateFluent<TEntity>> GetAggregateAsync<TEntity>(this IBasicRepository<TEntity> repository, CancellationToken cancellationToken = default)
+            where TEntity : class, IEntity
         {
-            return repository.ToMongoDbRepository().GetAggregateAsync(cancellationToken);
```

Comment:
There is a recursive call because the `IMongoDbRepository` does not have the `GetAggregateAsync` method.

Reply:
See https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.MongoDB/Volo/Abp/Domain/Repositories/MongoDB/IMongoDbRepository.cs#L28

Reply:
Sorry, I tested the code with 4.2.1 library so I missed the new feature. I will rollback it.